pkg/sources/sources.go

```diff
@@ -104,39 +105,75 @@ func NewConfig(opts ...func(*Config)) Config {
 	return *c
 }
 
+// Counter is a simple monotonically-increasing counter for tracking progress.
+// It includes a successCnt that records the number of successful scans.
+// It also includes a totalCnt that records the total number of scans.
+// This is necessary in order to display the number of concurrent scans
+// regardless of whether they are successful or not.
+// successCnt is the only field used when calculating the percentage complete.
+type Counter struct {
+	successCnt, totalCnt uint32
+	mu                   sync.Mutex
+}
+
+// IncTotal increments the total field of counter and returns the new value.
+func (c *Counter) IncTotal() int {
+	return int(atomic.AddUint32(&c.totalCnt, 1))
+}
+
+// IncSuccess increments the success field of counter and returns the new value.
+func (c *Counter) IncSuccess() int {
+	return int(atomic.AddUint32(&c.successCnt, 1))
+}
+
+// Get returns the current successCnt.
+func (c *Counter) Get() uint32 {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.successCnt
```

Comment:
**Nit (optional):** There's also a [LoadUint32](https://pkg.go.dev/sync/atomic#LoadUint32) function if you want to get rid of the mutex.